scripts/src/primazatest/runtest.py

```diff
@@ -0,0 +1,195 @@
+import argparse
+import subprocess
+import sys
+import time
+
+
+def run_cmd(cmd, silent=False):
+
+    if not silent:
+        curr_time = time.strftime("%I:%M:%S %p", time.localtime())
+        print(f"\n,---------,-\n| COMMAND: {curr_time} :"
+              f" {' '.join(cmd)}\n'---------'-")
+
+    ctl_out = subprocess.run(cmd, capture_output=True)
+
+    out = ""
+    if ctl_out.stdout:
+        out = ctl_out.stdout.decode("utf-8")
+
+    err = ""
+    if ctl_out.stderr:
+        err = ctl_out.stderr.decode("utf-8").strip()
+
+    return out, err
+
+
+def run_and_check(venv_dir, args, expect_msg, expect_error_msg, fail_msg):
+
+    command = [f"{venv_dir}/bin/primazactl"]
+    if args:
+        command += args
+    ctl_out, ctl_err = run_cmd(command)
+
+    outcome = True
+
+    if expect_msg:
+        if ctl_out:
+            print(f"Response was:\n{ctl_out}")
+            if expect_msg in ctl_out:
+                print(f"\n+++[PASS] args: {args}\n")
+            else:
+                print(f"\n---[FAIL] args: {args} : {fail_msg}\n")
+                outcome = False
+        else:
+            print(f"\n---[FAIL] args: {args} : {fail_msg}\n")
+            outcome = False
+
+    if expect_error_msg:
+        if ctl_err:
+            print(f"Error response was:\n{ctl_err}\n")
+            if expect_error_msg in ctl_err:
+                print(f"\n+++[PASS] args: {args}\n")
+            else:
+                print(f"\n---[FAIL] args: {args} : {fail_msg}\n")
+                outcome = False
+        else:
+            print(f"\n---[FAIL] args: {args} : {fail_msg}\n")
+            outcome = False
+
+    return outcome
+
+
+def test_args(venv_dir):
```

Comment:
Long-term, we should probably use [`pytest`](https://pypi.org/project/pytest/) for unit tests.

Reply:
Long term we plan to use behave. This is just a quick and dirty for now.